### PR TITLE
Add log warning on challenge creation

### LINKED ISSUE #1668: Trigger challenge fails if android app is deinstalled
# What did you try to do?

* Enrolled push token with android app
* Removed android app (push token is still in privacyIDEA)
* validate/triggerchallenge

# What outcome did you expect?

Trigger challenge shows my tokens and their messages

# What outcome did you experience?

HTTP 400 BAD REQUEST
result -> error -> code: 401
result -> error -> message: "ERR401: Failed to submit message to firebase service."

# Configuration

* privacyIDEA Version: 3.0.1 (master branch)
* Installation method: venv

## more details: 

* OS: Ubuntu 19.04
* Webserver: Apache2
* Tokendatabase: MySQL
* Android: latest beta from play store (2.0.1?)

# Log file

The attached log file is the response after I did /validate/triggerchallange 

This repeats 134 times with different lines (see log file)
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/logging/handlers.py", line 77, in emit
    self.doRollover()
  File "/usr/lib/python2.7/logging/handlers.py", line 142, in doRollover
    os.rename(self.baseFilename, dfn)
OSError: [Errno 13] Permission denied
Logged from file before_after.py, line 88
```
This is the last line
```
<my pi ip> - - [11/Jun/2019 12:39:29] "POST /validate/triggerchallenge HTTP/1.1" 400 -
```

[privacyidea.log](https://github.com/privacyidea/privacyidea/files/3276051/privacyidea.log)

### PR DESCRIPTION
**Pull request by timosturm**
Add a warning to the log if submitting the challenge to firebase fails.
Closes #1668 and #2738

### CODE REVIEW

File: privacyidea/lib/tokens/pushtoken.py [hunk 1]
```diff
@@ -860,9 +860,11 @@ def create_challenge(self, transactionid=None, options=None):
                 db_challenge.save()
                 self.challenge_janitor()
 
-            # If sending the Push message failed, we still raise an error.
+            # If sending the Push message failed, we still raise an error and a warning.
             if not res:
-                raise ValidateError("Failed to submit message to firebase service.")
+                log.warning("Failed to submit message to firebase service for token {0!s}."
+                            .format(self.token.serial))
+                raise ValueError("Failed to submit message to firebase service.")
```
Comment (cornelinux) [inline privacyidea/lib/tokens/pushtoken.py:867]: Why did you change this to ValueError?

**Review by cornelinux — CHANGES_REQUESTED** [review]
Please check, why the test is failing.

File: privacyidea/lib/tokens/pushtoken.py (hunk 1 above)
Comment (timosturm) [reply privacyidea/lib/tokens/pushtoken.py:867]: This really shouldn't be here because I already caught that one. I will fix it immediately.

File: privacyidea/lib/tokens/pushtoken.py [hunk 2]
```diff
@@ -860,8 +860,10 @@ def create_challenge(self, transactionid=None, options=None):
                 db_challenge.save()
                 self.challenge_janitor()
 
-            # If sending the Push message failed, we still raise an error.
+            # If sending the Push message failed, we still raise an error and a warning.
             if not res:
+                log.warning("Failed to submit message to firebase service for token {0!s}."
+                            .format(self.token.serial))
```
Comment (cornelinux) [inline privacyidea/lib/tokens/pushtoken.py:866]: I am sorry for my late comment.
I was not quite sure about this.

But in theory a serial number could contain a special character or a German umlaut. If we print it in the log message this way, we will raise an exception.
So we should do

    log.warnung(u"Failed to .....".format())

To allow logging unicode.

Thanks a lot

**Review by cornelinux — CHANGES_REQUESTED** [review]
Sorry. One last comment

File: privacyidea/lib/tokens/pushtoken.py (hunk 2 above)
Comment (timosturm) [reply privacyidea/lib/tokens/pushtoken.py:866]: This should be fixed now.

**Review by cornelinux — APPROVED** [review]
